src/CloudNative.CloudEvents.Amqp/AmqpClientExtensions.cs

```diff
@@ -21,26 +25,48 @@ public static class AmqpClientExtensions
 
-        public static CloudEvent ToCloudEvent(this Message message,
+        /// <summary>
+        /// Converts this AMQP message into a CloudEvent object.
+        /// </summary>
+        /// <param name="message">The AMQP message to convert. Must not be null.</param>
+        /// <param name="formatter">The event formatter to use to parse the CloudEvent. Must not be null.</param>
+        /// <param name="extensionAttributes">The extension attributes to use when parsing the CloudEvent. May be null.</param>
+        /// <returns>A reference to a validated CloudEvent instance.</returns>
+        public static CloudEvent ToCloudEvent(
+            this Message message,
             CloudEventFormatter formatter,
-            params CloudEventAttribute[] extensionAttributes)
+            params CloudEventAttribute[] extensionAttributes) =>
+            ToCloudEvent(message, formatter, (IEnumerable<CloudEventAttribute>) extensionAttributes);
+
+        /// <summary>
+        /// Converts this AMQP message into a CloudEvent object.
+        /// </summary>
+        /// <param name="httpResponseMessage">The AMQP message to convert. Must not be null.</param>
+        /// <param name="formatter">The event formatter to use to parse the CloudEvent. Must not be null.</param>
+        /// <param name="extensionAttributes">The extension attributes to use when parsing the CloudEvent. May be null.</param>
+        /// <returns>A reference to a validated CloudEvent instance.</returns>
+        public static CloudEvent ToCloudEvent(
+            this Message message,
+            CloudEventFormatter formatter,
+            IEnumerable<CloudEventAttribute> extensionAttributes)
         {
+            message = message ?? throw new ArgumentNullException(nameof(message));
+            formatter = formatter ?? throw new ArgumentNullException(nameof(formatter));
+
             if (HasCloudEventsContentType(message, out var contentType))
             {
                 return formatter.DecodeStructuredModeMessage(new MemoryStream((byte[]) message.Body), new ContentType(contentType), extensionAttributes);
             }
             else
             {
                 var propertyMap = message.ApplicationProperties.Map;
-                if (!propertyMap.TryGetValue(SpecVersionAmqpHeader, out var versionId) || !(versionId is string versionIdText))
+                if (!propertyMap.TryGetValue(SpecVersionAmqpHeader, out var versionId))
                 {
                     throw new ArgumentException("Request is not a CloudEvent");
                 }
-                var version = CloudEventsSpecVersion.FromVersionId(versionIdText);
-                if (version is null)
-                {
-                    throw new ArgumentException($"Unsupported CloudEvents spec version '{versionIdText}'");
-                }
+
+                var version = CloudEventsSpecVersion.FromVersionId(versionId as string)
```

Comment:
Should we have a null check for versionId? Otherwise this would throw a NRE I believe.